smartfs.lua

```diff
@@ -263,13 +272,27 @@ function smartfs._show_(form, player, params, is_inv)
 	return state
 end
 
+-- Attach a formspec to a node
+function smartfs._attach_nodemeta_(form, nodepos, placer)
+	local state = smartfs._makeState_(form, placer, params, nil, nodepos)
+	if form._reg(state) then
+		state:_attach_nodemeta_()
+	end
+	return state
+end
+
 -- Receive fields from formspec
 local function _sfs_recieve_(state,name,fields)
 	if (fields.quit == "true") then
-		if not state.is_inv then
+		if state.nodepos then
+			state:close() --mark as closed for this user (will be reopened if other user is on the node)
```

Comment:
It is a hack, in current implementation the "closed" is reset in smartfs.nodemeta_on_receive_fields after the call if other users are connected (line 368) `state.closed = nil --not closed`

You are right, the viewers count will be better. Or just reading of next(state.multiplayer)
Should I rename state.multiplayer to something other? My proposal to change the state.player from currently "playername" to players table and replace the multiplayer table